### PR TITLE
Add store for products finder taxons scope

### PR DESCRIPTION
**Pull request by szymoniwacz**

### CODE REVIEW

File: api/app/controllers/spree/api/v2/storefront/products_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def sorted_collection
           end
 
           def collection
-            @collection ||= collection_finder.new(scope: scope, params: finder_params).execute
+            @collection ||= collection_finder.new(scope: scope, params: finder_params, store: current_store).execute
```
Comment (damianlegawiec) [inline api/app/controllers/spree/api/v2/storefront/products_controller.rb:13]: @szymoniwacz `finder_params` already includes `store` - https://github.com/spree/spree/blob/master/api/app/controllers/spree/api/v2/resource_controller.rb#L58

**Review by tomurb — APPROVED** [review]

File: core/app/finders/spree/products/find.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ def initialize(scope:, params:, current_currency: nil)
 
         @ids              = String(params.dig(:filter, :ids)).split(',')
         @skus             = String(params.dig(:filter, :skus)).split(',')
+        @store            = params[:store]
```
Comment (damianlegawiec) [inline core/app/finders/spree/products/find.rb:18]:
```suggestion
        @store            = params[:store] || Spree::Store.default
```

File: api/app/serializers/spree/v2/storefront/product_serializer.rb [hunk 3]
```diff
@@ -47,7 +47,10 @@ class ProductSerializer < BaseSerializer
         has_many :variants
         has_many :option_types
         has_many :product_properties
-        has_many :taxons
+
+        has_many :taxons do |object, params|
+          params[:store].taxons.joins(:products).where(products: { id: object.id })
```
Comment (damianlegawiec) [inline api/app/serializers/spree/v2/storefront/product_serializer.rb:52]:
```suggestion
          object.taxons.for_store(params[:store])
```

let's move this into the model as we're going to re-use this :) 

in Taxon model:

```ruby
scope :for_store, ->(store) { joins(:taxonomy).where(spree_taxonomies: { store_id: store.id }) }
```

Comment (szymoniwacz) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb:52]: @damianlegawiec I can do that, but it is overriding default `for_store` scope. https://github.com/spree/spree/blob/master/core/app/models/spree/base.rb#L25

Comment (damianlegawiec) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb]: @szymoniwacz sorry my bad, we don't need it :)

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**